core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -80,6 +80,7 @@ public TransactionOutPoint(long index, Transaction fromTx) {
         this.index = index;
         this.hash = fromTx.getTxId();
         this.fromTx = fromTx;
+        this.connectedOutput = Objects.requireNonNull(fromTx.getOutput(index));
```

Comment:
This should make line 96 unnecessary (or maybe converted to an assert rather than an assignment).

Reply:
Also, the first if branch in `getConnectedOutput()` is potentially unnecessary after this change?

`if (fromTx != null) `